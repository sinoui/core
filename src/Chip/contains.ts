/**
 * 标准模式下放人背景色
 */
export const BASIC_CHIP_BG = {
  light: '#e0e0e0',
  dark: '#616161',
};

/**
 * 边框线的默认颜色
 */
export const BORDER_COLOR = {
  light: 'rgba(0,0,0,0.23)',
  dark: 'rgba(255,255,255,0.23)',
};

/**
 * 删除图标颜色
 */
export const DELETE_ICON_COLOR = {
  light: 'rgba(0,0,0,0.26)',
  dark: 'rgba(255,255,255,0.26)',
};

/**
 * 删除图标hover颜色
 */
export const DELETE_ICON_HOVER_COLOR = {
  light: 'rgba(0,0,0,0.4)',
  dark: 'rgba(255,255,255,0.4)',
};

/**
 * 前缀图标颜色
 */
export const LEADING_ICON_COLOR = {
  light: '#616161',
  dark: '#e0e0e0',
};
