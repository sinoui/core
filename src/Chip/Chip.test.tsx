import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import 'jest-styled-components';
import Chip from './Chip';

describe('镜像测试', () => {
  it('基本渲染', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <>
            <Chip label="文本" />
            <Chip label="文本" variant="outlined" />
          </>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('带删除图标', () => {
    const onDelete = jest.fn();
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <>
            <Chip label="文本" onDelete={onDelete} />
            <Chip label="文本" variant="outlined" onDelete={onDelete} />
          </>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('不可用', () => {
    const onDelete = jest.fn();
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <>
            <Chip label="文本" onDelete={onDelete} disabled />
            <Chip
              label="文本"
              variant="outlined"
              onDelete={onDelete}
              disabled
            />
          </>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('可点击的标签', () => {
    const onDelete = jest.fn();
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <>
            <Chip label="文本" clickable />
            <Chip label="文本" clickable variant="outlined" />
            <Chip label="文本" onDelete={onDelete} clickable />
            <Chip
              label="文本"
              variant="outlined"
              onDelete={onDelete}
              clickable
            />
          </>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('密集模式', () => {
    const onDelete = jest.fn();
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <>
            <Chip label="文本" dense />
            <Chip label="文本" dense variant="outlined" />
            <Chip label="文本" onDelete={onDelete} dense />
          </>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('验收测试', () => {
  afterEach(cleanup);

  it('点击删除按钮，onDelete被调用', () => {
    const onDelete = jest.fn();
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Chip label="文本" onDelete={onDelete} />
      </ThemeProvider>,
    );

    const clearButton = container.querySelector(
      '.sinoui-chip__delete',
    ) as HTMLElement;

    fireEvent.click(clearButton);

    expect(onDelete).toHaveBeenCalled();
  });

  it('禁用时，点击清除按钮没有作用', () => {
    const onDelete = jest.fn();
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Chip label="文本" onDelete={onDelete} disabled />
      </ThemeProvider>,
    );

    const clearButton = container.querySelector(
      '.sinoui-chip__delete',
    ) as HTMLElement;

    fireEvent.click(clearButton);

    expect(onDelete).not.toHaveBeenCalled();
  });

  it('variant=outlined', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={defaultTheme}>
        <Chip label="文本" variant="outlined" data-testid="chip" />
      </ThemeProvider>,
    );

    expect(getByTestId('chip')).toHaveClass('sinoui-chip--outlined');
  });

  it('指向根元素的属性', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={defaultTheme}>
        <Chip
          label="文本"
          variant="outlined"
          data-testid="chip"
          id="chip"
          className="custom-chip"
        />
      </ThemeProvider>,
    );

    expect(getByTestId('chip')).toHaveClass('custom-chip');
    expect(getByTestId('chip')).toHaveAttribute('id', 'chip');
  });

  it('ref 指向根元素', () => {
    const ref = React.createRef<HTMLDivElement>();
    const { getByTestId } = render(
      <ThemeProvider theme={defaultTheme}>
        <Chip label="文本" data-testid="chip" ref={ref} />
      </ThemeProvider>,
    );

    expect(ref.current).toBe(getByTestId('chip'));
  });
});

it('密集模式', () => {
  const { getByTestId } = render(
    <ThemeProvider theme={defaultTheme}>
      <Chip
        label="文本"
        dense
        data-testid="chip"
        onDelete={() => console.log('123')}
      />
    </ThemeProvider>,
  );
  expect(getByTestId('chip')).toHaveStyleRule('height', '24px');
});
