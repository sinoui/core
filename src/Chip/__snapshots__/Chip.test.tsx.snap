// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`镜像测试 不可用 1`] = `
Array [
  .c1 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.38);
  opacity: 0.5;
}

.c2 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: default;
  color: rgba(0,0,0,0.38);
}

.c2:hover {
  fill: rgba(0,0,0,0.38);
}

@media (hover:none) {
  .c2:hover {
    fill: rgba(0,0,0,0.38);
  }
}

<div
    aria-disabled="true"
    className="c0 sinoui-chip sinoui-chip--disabled"
    disabled={true}
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c1 c2 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c2 sinoui-chip__delete"
      disabled={true}
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>,
  .c1 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.38);
  border: 1px solid rgba(0,0,0,0.23);
  background-color: transparent;
  opacity: 0.5;
}

.c2 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: default;
  color: rgba(0,0,0,0.38);
}

.c2:hover {
  fill: rgba(0,0,0,0.38);
}

@media (hover:none) {
  .c2:hover {
    fill: rgba(0,0,0,0.38);
  }
}

<div
    aria-disabled="true"
    className="c0 sinoui-chip sinoui-chip--disabled sinoui-chip--outlined"
    disabled={true}
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c1 c2 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c2 sinoui-chip__delete"
      disabled={true}
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>,
]
`;

exports[`镜像测试 可点击的标签 1`] = `
Array [
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0px;
  padding: 0px;
  margin: 0;
  border-radius: 0;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  pointer-events: default;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0:hover {
  background-color: rgba(0,0,0,0.04);
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
}

.c1:hover {
  background-color: rgba(0,0,0,0.26);
}

@media (hover:none) {
  .c0:hover {
    background-color: transparent;
  }
}

@media (hover:none) {

}

<button
    className="c0 sinoui-base-button c1 sinoui-chip"
    role="button"
    type="button"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
  </button>,
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0px;
  padding: 0px;
  margin: 0;
  border-radius: 0;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  pointer-events: default;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0:hover {
  background-color: rgba(0,0,0,0.04);
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  border: 1px solid rgba(0,0,0,0.23);
  background-color: transparent;
}

@media (hover:none) {
  .c0:hover {
    background-color: transparent;
  }
}

@media (hover:none) {

}

<button
    className="c0 sinoui-base-button c1 sinoui-chip sinoui-chip--outlined"
    role="button"
    type="button"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
  </button>,
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0px;
  padding: 0px;
  margin: 0;
  border-radius: 0;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  pointer-events: default;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0:hover {
  background-color: rgba(0,0,0,0.04);
}

.c2 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
}

.c1:hover {
  background-color: rgba(0,0,0,0.26);
}

.c3 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: pointer;
  color: rgba(0,0,0,0.6);
}

.c3:hover {
  fill: rgba(0,0,0,0.87);
}

@media (hover:none) {
  .c0:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c3:hover {
    fill: rgba(0,0,0,0.6);
  }
}

<button
    className="c0 sinoui-base-button c1 sinoui-chip"
    role="button"
    type="button"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c2 c3 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c3 sinoui-chip__delete"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </button>,
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0px;
  padding: 0px;
  margin: 0;
  border-radius: 0;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  pointer-events: default;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0:hover {
  background-color: rgba(0,0,0,0.04);
}

.c2 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  border: 1px solid rgba(0,0,0,0.23);
  background-color: transparent;
}

.c3 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: pointer;
  color: rgba(0,0,0,0.6);
}

.c3:hover {
  fill: rgba(0,0,0,0.87);
}

@media (hover:none) {
  .c0:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c3:hover {
    fill: rgba(0,0,0,0.6);
  }
}

<button
    className="c0 sinoui-base-button c1 sinoui-chip sinoui-chip--outlined"
    role="button"
    type="button"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c2 c3 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c3 sinoui-chip__delete"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </button>,
]
`;

exports[`镜像测试 基本渲染 1`] = `
Array [
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
}

<div
    className="c0 sinoui-chip"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
  </div>,
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  border: 1px solid rgba(0,0,0,0.23);
  background-color: transparent;
}

<div
    className="c0 sinoui-chip sinoui-chip--outlined"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
  </div>,
]
`;

exports[`镜像测试 密集模式 1`] = `
Array [
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  height: 24px;
}

.c0 > .sinoui-chip__delete {
  width: 20px;
  height: 20px;
  margin-right: -8px;
}

@media (hover:none) {

}

<div
    className="c0 sinoui-chip"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
  </div>,
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  border: 1px solid rgba(0,0,0,0.23);
  background-color: transparent;
  height: 24px;
}

.c0 > .sinoui-chip__delete {
  width: 20px;
  height: 20px;
  margin-right: -8px;
}

@media (hover:none) {

}

<div
    className="c0 sinoui-chip sinoui-chip--outlined"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
  </div>,
  .c1 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  height: 24px;
}

.c0 > .sinoui-chip__delete {
  width: 20px;
  height: 20px;
  margin-right: -8px;
}

.c2 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: pointer;
  color: rgba(0,0,0,0.6);
}

.c2:hover {
  fill: rgba(0,0,0,0.87);
}

@media (hover:none) {
  .c2:hover {
    fill: rgba(0,0,0,0.6);
  }
}

<div
    className="c0 sinoui-chip"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c1 c2 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c2 sinoui-chip__delete"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>,
]
`;

exports[`镜像测试 带删除图标 1`] = `
Array [
  .c1 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
}

.c2 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: pointer;
  color: rgba(0,0,0,0.6);
}

.c2:hover {
  fill: rgba(0,0,0,0.87);
}

@media (hover:none) {
  .c2:hover {
    fill: rgba(0,0,0,0.6);
  }
}

<div
    className="c0 sinoui-chip"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c1 c2 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c2 sinoui-chip__delete"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>,
  .c1 {
  font-size: 24px;
  width: 1em;
  height: 1em;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  border-radius: 16px;
  padding: 0 12px;
  overflow: hidden;
  background-color: #e0e0e0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
  color: rgba(0,0,0,0.87);
  border: 1px solid rgba(0,0,0,0.23);
  background-color: transparent;
}

.c2 {
  margin-left: 8px;
  margin-right: -4px;
  cursor: pointer;
  color: rgba(0,0,0,0.6);
}

.c2:hover {
  fill: rgba(0,0,0,0.87);
}

@media (hover:none) {
  .c2:hover {
    fill: rgba(0,0,0,0.6);
  }
}

<div
    className="c0 sinoui-chip sinoui-chip--outlined"
  >
    <span
      className="sinoui-chip__content"
    >
      文本
    </span>
    <svg
      aria-hidden="true"
      className="c1 c2 sinoui-chip__delete sinoui-svg-icon sc-AxgMl c2 sinoui-chip__delete"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      size={24}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>,
]
`;
